src/views/pages/Inbox/index.js

```diff
@@ -1,19 +1,24 @@
 import React, {useEffect, useState} from 'react';
-import MessageService from '../../../services/Message.service';
+import MessageAPIService from '../../../services/MessageAPI';
 import ListView from '../../components/ListView';
 
-const message = new MessageService();
+const message = new MessageAPIService();
 
 function Inbox() {
   const [messageList, setMessageList] = useState([]);
 
   useEffect(() => {
-    const sub = message.messages$.subscribe((d) => {
+    const sub = message.received_messages$.subscribe((d) => {
       setMessageList(d);
     });
 
+    // TODO checkMessages needs to pull its argument from our current identity.
+    // TODO Why is it pulling messages twice?
+    let id = setInterval(message.checkMessages(1), 5000);
```

Comment:
this fixes the double hit on the API
```suggestion
    let id = setInterval(() => {
       message.checkMessages(1);
    }, 5000);
```